### PR TITLE
tbls: support deterministic key gen for testing

### PR DESCRIPTION
**Pull request by corverroos**
Adds support for deterministic tbls key generation for use in golden file tests.

category: test
ticket: none

### CODE REVIEW

File: cluster/test_cluster.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ func NewForT(t *testing.T, dv, k, n, seed int, opts ...func(*Definition)) (Lock,
 	}
 
 	for i := 0; i < dv; i++ {
-		rootSecret, err := tbls.GenerateSecretKey()
+		rootSecret, err := tbls.GenerateInsecureKey(t, random)
```
Comment (corverroos) [inline cluster/test_cluster.go:42]: the clusters generated by this function should be deterministic.

File: tbls/herumi.go [hunk 2]
```diff
@@ -36,6 +39,27 @@ func init() {
...
+func (Herumi) GenerateInsecureKey(t *testing.T, random io.Reader) (PrivateKey, error) {
+	t.Helper()
+	for i := 0; i < 100; i++ {
+		b := make([]byte, 32)
+		_, err := random.Read(b)
+		require.NoError(t, err)
+
+		p := new(bls.SecretKey)
+		err = p.Deserialize(b)
+		if err != nil {
```
Comment (corverroos) [inline tbls/herumi.go:53]: not all random values are valid, seems around 50% are valid.

Comment (gsora) [reply tbls/herumi.go:53]: AFAIK `bls.SetRandFunc` can be used to tell the Herumi library to use a specific source of entropy.

Might help with this issue?

File: tbls/tbls.go [hunk 3]
```diff
@@ -25,6 +29,10 @@ type Implementation interface {
 	// GenerateSecretKey generates a secret key and returns its compressed serialized representation.
 	GenerateSecretKey() (PrivateKey, error)
 
+	// GenerateInsecureKey generates a insecure deterministic secret key using the provided random number generator
```
Comment (xenowits) [inline tbls/tbls.go:32]: nit:
```suggestion
	// GenerateInsecureKey generates an insecure deterministic secret key using the provided random number generator
```

**Review by xenowits — APPROVED** [review]

**Pull request merged**